frontend/src/modules/SeismicIntersection/view.tsx

```diff
@@ -0,0 +1,215 @@
+        if (!isEqual(newExtendedWellboreTrajectory, extendedWellboreTrajectory)) {
+            setExtendedWellboreTrajectory(newExtendedWellboreTrajectory);
+
+            const x_points = newExtendedWellboreTrajectory?.points.map((coord) => coord[0]) ?? [];
+            const y_points = newExtendedWellboreTrajectory?.points.map((coord) => coord[1]) ?? [];
+            candidateSeismicFencePolyline = { x_points, y_points };
+            setSeismicFencePolyline(candidateSeismicFencePolyline);
+        }
+
+        // When new well trajectory 3D points are loaded, update the render trajectory and clear the seismic fence image
+        if (!isEqual(trajectoryXyzPoints, renderWellboreTrajectoryXyzPoints)) {
+            setRenderWellboreTrajectoryXyzPoints(trajectoryXyzPoints);
+            setGenerateSeismicSliceImageOptions(null);
+        }
+    }
+
+    // Get seismic fence data from polyline
+    const seismicFenceDataQuery = useSeismicFenceDataQuery(
+        seismicAddress?.caseUuid ?? null,
+        seismicAddress?.ensemble ?? null,
+        seismicAddress?.realizationNumber ?? null,
+        seismicAddress?.attribute ?? null,
+        seismicAddress?.timeString ?? null,
+        seismicAddress?.observed ?? null,
+        candidateSeismicFencePolyline,
+        seismicAddress !== null
+    );
+    if (seismicFenceDataQuery.isError) {
+        statusWriter.addError("Error loading seismic fence data");
+    }
+
+    if (seismicFenceDataQuery.data) {
+        // Get an array of projected 2D points [x, y], as 2D curtain projection from a set of trajectory 3D points and offset
+        const newExtendedWellboreTrajectoryXyProjection: number[][] = extendedWellboreTrajectory
+            ? IntersectionReferenceSystem.toDisplacement(
+                  extendedWellboreTrajectory.points,
+                  extendedWellboreTrajectory.offset
+              )
+            : [];
+
+        const newSeismicImageDataArray = createSeismicSliceImageDataArrayFromFenceData(seismicFenceDataQuery.data);
+        const newSeismicImageYAxisValues = createSeismicSliceImageYAxisValuesArrayFromFenceData(
+            seismicFenceDataQuery.data
+        );
+
+        const newGenerateSeismicSliceImageOptions: SeismicSliceImageOptions = {
+            dataValues: newSeismicImageDataArray,
+            yAxisValues: newSeismicImageYAxisValues,
+            trajectoryXyPoints: newExtendedWellboreTrajectoryXyProjection,
+            colormap: seismicColors,
+            extension: extension,
+        };
+
+        if (!isEqual(generateSeismicSliceImageOptions, newGenerateSeismicSliceImageOptions)) {
+            setGenerateSeismicSliceImageOptions(newGenerateSeismicSliceImageOptions);
+        }
+    }
+
+    // Update esv-intersection controller when data is ready - keep old data to prevent blank view when fetching new data
+    if (esvIntersectionControllerRef.current && renderWellboreTrajectoryXyzPoints) {
```

Comment:
Consider improving this in the future as it might be beneficial to not call this whole bunch on every render when the actual data used has not really been changed.

Reply:
I'll create a separate issue on this. To remember.